src/Bridges/bridge_optimizer.jl

```diff
@@ -1347,14 +1347,44 @@ function MOI.supports(
     attr::MOI.AbstractConstraintAttribute,
     IndexType::Type{MOI.ConstraintIndex{F,S}},
 ) where {F,S}
-    if is_bridged(b, F, S)
-        bridge = Constraint.concrete_bridge_type(b, F, S)
-        return MOI.supports(recursive_model(b), attr, bridge)
-    elseif F == MOI.Utilities.variable_function_type(S) && is_bridged(b, S)
-        bridge = Variable.concrete_bridge_type(b, S)
-        return MOI.supports(recursive_model(b), attr, bridge)
+    # !!! warning
+    #     This function is slightly confusing, because we need to account for
+    #     the different ways in which a constraint might be added.
+    is_variable = F == MOI.Utilities.variable_function_type(S)
+    if is_variable
+        # There are three ways an `is_variable` constraint might be added:
+        #  1. It is supported natively
+        #  2. It is added as a free variable, followed by a constraint bridge
+        #  3. It is added by a variable bridge
+        # To distinguish between {1.} and {2., 3.}, we call `is_bridged`:
+        if is_bridged(b, S)
```

Comment:
Suppose you have a solver that supports
* Adding variables constrained on creation in S or T
* Adding a constraint VAF-in-T

You add variables on creation to S
Then you add a constraint on this vector of variables in T. This is bridged into VAF-in-T.
Here, `is_bridged(b, T)` is `false` with a `LazyBridgeOptimizer`.

There are cases in which both might be possible, it depends whether it was constrained on creation or not. In that case, `supports` should return `true` if one of the two cases is `true` because there is a chance that it is supported.

Reply:
I'm going around in circles. Is there a way to tell if a `VectorOfVariables` was added by `add_constrained_variables`? Not really just from the type, right?

Reply:
So I think we actually have a much bigger issue that means we can't get this right.

The `SingleBridgeOptimizer` tests are failing for things like `flip_sign.jl`, because if you add a `VectorOfVariables-in-Nonnegatives` constraint, the solver supports the constraint and the attribute. But if you add constrained variables in `Nonnegatives`, the solver force bridges (even though the model supports it) and the bridge doesn't support the attribute.

But, we can't distinguish whether to return `true` or `false`, because given a constraint index we don't know whether it was added as a constraint or a constrained variable.

Reply:
Exactly, given just the type, you cannot know how all constraint of that type will be dealt with.You need to consider all possible cases and return `true` if at least one case supports it